controllers/admin/AdminShoppingfeedAccountSettings.php

```diff
@@ -319,6 +352,7 @@ public function postProcess()
     public function saveToken($shop_id, $lang_id, $currency_id)
     {
         $token = Tools::getValue(Shoppingfeed::AUTH_TOKEN);
+        $store_id = Tools::getValue('store_id', '');
```

Comment:
store_id should be an integer according to the object model definition.
You probably should cast it or add a validation before using it.

Reply:
the modifications were added